File: backend/src/test/java/lan/dk/podcastserver/config/WebMvcConfigTest.java
package lan.dk.podcastserver.config;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.runners.MockitoJUnitRunner;
import org.springframework.web.servlet.config.annotation.DefaultServletHandlerConfigurer;
import org.springframework.web.servlet.config.annotation.ResourceChainRegistration;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistration;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;

import static org.mockito.Matchers.anyInt;
import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.*;

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@RunWith(MockitoJUnitRunner.class)
public class WebMvcConfigTest {

    @InjectMocks WebMvcConfig webMvcConfig;

    @Test
    public void should_add_resource_handlers() {
        /* Given */
        ResourceHandlerRegistry registry = mock(ResourceHandlerRegistry.class);
        ResourceHandlerRegistration resourceHandlerRegistration = mock(ResourceHandlerRegistration.class);
        ResourceChainRegistration resourceChainRegistration = mock(ResourceChainRegistration.class);

        when(registry.addResourceHandler(anyVararg())).thenReturn(resourceHandlerRegistration);
        when(resourceHandlerRegistration.addResourceLocations(anyString())).thenReturn(resourceHandlerRegistration);
        when(resourceHandlerRegistration.setCachePeriod(anyInt())).thenReturn(resourceHandlerRegistration);
        when(resourceHandlerRegistration.resourceChain(anyBoolean())).thenReturn(resourceChainRegistration);
        when(resourceChainRegistration.addResolver(any())).thenReturn(resourceChainRegistration);

        /* When */
        webMvcConfig.addResourceHandlers(registry);

        /* Then */
        verify(registry, times(2)).addResourceHandler(anyVararg());
    }

    @Test
    public void should_configure_servlet_handling() {
        /* Given */
        DefaultServletHandlerConfigurer configurer = mock(DefaultServletHandlerConfigurer.class);

        /* When */
        webMvcConfig.configureDefaultServletHandling(configurer);

        /* Then */
        verify(configurer, times(1)).enable();
    }
}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/config/WebMvcConfigTest.kt
package com.github.davinkevin.podcastserver.config

import com.github.davinkevin.podcastserver.config.WebMvcConfig
import com.nhaarman.mockitokotlin2.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.web.servlet.config.annotation.DefaultServletHandlerConfigurer
import org.springframework.web.servlet.config.annotation.ResourceChainRegistration
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistration
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@ExtendWith(MockitoExtension::class)
class WebMvcConfigTest {

    @InjectMocks lateinit var webMvcConfig: WebMvcConfig

    @Test
    fun should_add_resource_handlers() {
        /* Given */
        val registry = mock<ResourceHandlerRegistry>()
        val resourceHandlerRegistration = mock<ResourceHandlerRegistration>()
        val resourceChainRegistration = mock<ResourceChainRegistration>()

        whenever(registry.addResourceHandler(anyVararg())).thenReturn(resourceHandlerRegistration)
        whenever(resourceHandlerRegistration.addResourceLocations(any())).thenReturn(resourceHandlerRegistration)
        whenever(resourceHandlerRegistration.setCachePeriod(any())).thenReturn(resourceHandlerRegistration)
        whenever(resourceHandlerRegistration.resourceChain(any())).thenReturn(resourceChainRegistration)
        whenever(resourceChainRegistration.addResolver(any())).thenReturn(resourceChainRegistration)

        /* When */
        webMvcConfig.addResourceHandlers(registry)

        /* Then */
        verify(registry, times(2)).addResourceHandler(anyVararg())
    }

    @Test
    fun should_configure_servlet_handling() {
        /* Given */
        val configurer = mock<DefaultServletHandlerConfigurer>()

        /* When */
        webMvcConfig.configureDefaultServletHandling(configurer)

        /* Then */
        verify(configurer, times(1)).enable()
    }
}
